File: problems/500/solution.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        ans = []
        
        r1, r2, r3 = 'qQwWeErRtTyYuUiIoOpP', 'aAsSdDfFgGhHjJkKlL', 'zZxXcCvVbBnNmM'
        
        for word in words:
            if all([ch in r1 for ch in word]) or all([ch in r2 for ch in word]) or all([ch in r3 for ch in word]):
                ans.append(word)
        
        return ans

File: problems/500/solution.cpp
class Solution {
private:
    string rows[3] = {"qwertyuiop", "asdfghjkl", "zxcvbnm"};
    
public:
    vector<string> findWords(vector<string>& words) {
        vector<string> ans;
        unordered_map<char, int> m;
        
        for (int i = 0; i < 3; i++) {
            for (const char& ch: rows[i]) {
                m[tolower(ch)] = i;
                m[toupper(ch)] = i;
            }
        }
        
        for (const string& word: words) {
            unordered_set<int> count;
            
            for (const char& ch: word) {
                count.insert(m[ch]);
            }
            
            if (count.size() == 1) {
                ans.push_back(word);
            }
        }
        
        return ans;
    }
};